tensorflow/go/attrs.go

```diff
@@ -0,0 +1,258 @@
+			if err := status.Err(); err != nil {
+				return nil, err
+			}
+			vals := make([]int64, a.ListSize)
+			for i, val := range list {
+				vals[i] = int64(val)
+			}
+			return vals, nil
+
+		case FloatAttr:
+			list := make([]C.float, a.ListSize)
+			C.TF_OperationGetAttrFloatList(a.Op.c, cname, &list[0], C.int(a.ListSize), status.c)
+			if err := status.Err(); err != nil {
+				return nil, err
+			}
+			vals := make([]float32, a.ListSize)
+			for i, val := range list {
+				vals[i] = float32(val)
+			}
+			return vals, nil
+
+		case BoolAttr:
+			list := make([]C.uchar, a.ListSize)
+			C.TF_OperationGetAttrBoolList(a.Op.c, cname, &list[0], C.int(a.ListSize), status.c)
+			if err := status.Err(); err != nil {
+				return nil, err
+			}
+			vals := make([]bool, a.ListSize)
+			for i, val := range list {
+				vals[i] = val == 1
+			}
+			return vals, nil
+
+		case TypeAttr:
+			list := make([]C.TF_DataType, a.ListSize)
+			C.TF_OperationGetAttrTypeList(a.Op.c, cname, &list[0], C.int(a.ListSize), status.c)
+			if err := status.Err(); err != nil {
+				return nil, err
+			}
+			vals := make([]DataType, a.ListSize)
+			for i, val := range list {
+				vals[i] = DataType(val)
+			}
+			return vals, nil
+
+		case TensorAttr:
+			list := make([]*C.TF_Tensor, a.ListSize)
+			C.TF_OperationGetAttrTensorList(a.Op.c, cname, &list[0], C.int(a.ListSize), status.c)
+			if err := status.Err(); err != nil {
+				return nil, err
+			}
+			vals := make([]*Tensor, a.ListSize)
+			for i, t := range list {
+				vals[i] = newTensorFromC(t)
+			}
+			return vals, nil
+
+		case ShapeAttr:
+			dims := make([]*C.int64_t, a.ListSize)
+			numDims := make([]C.int, a.ListSize)
+			storage := make([]C.int64_t, a.TotalSize)
+			C.TF_OperationGetAttrShapeList(a.Op.c, cname, &dims[0], &numDims[0], C.int(a.ListSize), &storage[0], C.int(a.TotalSize), status.c)
+			if err := status.Err(); err != nil {
+				return nil, err
+			}
+			list := make([]Shape, a.ListSize)
+			for i, dim := range dims {
+				numDim := numDims[i]
+				// A []C.int64_t slice backed by C memory.
+				// See: https://github.com/golang/go/wiki/cgo#turning-c-arrays-into-go-slices
+				slice := (*[1 << 30]C.int64_t)(unsafe.Pointer(dim))[:numDim:numDim]
+				s := Shape{
+					dims: make([]int64, numDim),
+				}
+				for i, dim := range slice {
+					s.dims[i] = int64(dim)
+				}
+				list[i] = s
+			}
+			return list, nil
+
+		default:
+			return nil, errors.Errorf("list type %v not supported yet", a.Type)
+		}
+	}
+
+	switch a.Type {
+	case StringAttr:
+		v := make([]C.char, a.TotalSize)
+		C.TF_OperationGetAttrString(a.Op.c, cname, unsafe.Pointer(&v[0]), C.size_t(a.TotalSize), status.c)
+		if err := status.Err(); err != nil {
+			return nil, err
+		}
+		return C.GoStringN(&v[0], C.int(a.TotalSize)), nil
+
+	case IntAttr:
+		var v C.int64_t
+		C.TF_OperationGetAttrInt(a.Op.c, cname, &v, status.c)
+		if err := status.Err(); err != nil {
```

Comment:
Could you simplify this and many other lines below to:

```go
return int64(v), status.Err()
```

etc.?

Reply:
Done.